usvm-ts/src/main/kotlin/org/usvm/TSTypeStorage.kt

```diff
@@ -0,0 +1,28 @@
+package org.usvm
+
+import org.jacodb.ets.base.EtsType
+
+/*
+    This is a very basic implementation of type storage with memory and model objects interoperability.
```

Comment:
How it works? For the same state there are multiple possible models available. Why is it usefull?

Reply:
For different entities (register readings, lists, etc.) with ```addressSort``` it stores type suggestions from type coercion. Different models will have the same access path to their types as each type of ref entity has its own way of defining a key.

Reply:
For example, registet readings use its' ```idx``` as a key in TypeStorage

Reply:
Explanation enriched in the code, please check